klient/internal/testutil/setup.go

```diff
@@ -0,0 +1,90 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package testutil
+
+import (
+	"flag"
+	"k8s.io/client-go/util/homedir"
+	"log"
+	"path/filepath"
+	"time"
+
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"sigs.k8s.io/e2e-framework/klient/conf"
+	"sigs.k8s.io/e2e-framework/support/kind"
+)
+
+type TestCluster struct {
+	KindCluster *kind.Cluster
+	Kubeconfig string
+	RESTConfig *rest.Config
+	Clientset kubernetes.Interface
+}
+
+func SetupTestCluster(path string) *TestCluster {
+	if path == "" {
+		home := homedir.HomeDir()
+		path = filepath.Join(home, ".kube", "config")
```

Comment:
You may be interested in using `conf.ResolveKubeConfigFile` (found here https://github.com/kubernetes-sigs/e2e-framework/blob/main/klient/conf/config.go)

Reply:
Yes. I am using that already and like you mentioned below, I think we can remove Line 52-57 and this one too.